### PR TITLE
more than fun, you're the sanctuary

### PR DESCRIPTION
**Pull request by ryanchou-dev**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [ ] I have tested my code.
- [ ] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ ] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

**Comment by ryanchou-dev** [comment]
python took an L

File: solutions/usaco-989.mdx [hunk 1]
```diff
@@ -2,11 +2,215 @@
...
+  while(left>0) {
+    if(ok(speed+1, x, left-(speed+1))) {
+      // speed up if we can
+      speed += 1;
+    } else if(ok(speed, x, left-speed)) {
+      // otherwise keep the same speed if we can
+    } else {
+      // slow down if we have to
+      // check that we have enough space left to stop in time.
+      assert(ok(speed-1, x, left-(speed-1)));
```
Comment (SansPapyrus683) [inline solutions/usaco-989.mdx:61]: tabs

Comment (ryanchou-dev) [reply solutions/usaco-989.mdx:61]: oh bruh the auto-formatter didn't catch that

**Comment by SansPapyrus683** [comment]
Anta ga suki de sukisugite

**Pull request merged**